convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -1411,21 +1411,90 @@ def test_call_yum_cmd_w_downgrades(monkeypatch, retcode, output):
 @pytest.mark.parametrize(
     ("version1", "version2", "expected"),
     (
-        ("123-4.fc35", "123-4.fc35", 0),
-        ("123-5.fc35", "123-4.fc35", 1),
-        ("123-3.fc35", "123-4.fc35", -1),
+        ("kernel-core-0:4.18.0-240.10.1.el8_3.i86", "kernel-core-0:4.18.0-240.10.1.el8_3.i86", 0),
+        ("kernel-core-0:123-5.fc35", "kernel-core-0:123-4.fc35", 1),
+        ("kernel-core-123-3.fc35.aarch64", "kernel-core-123-4.fc35.aarch64", -1),
+        ("kernel-3.10.0-1160.83.1.0.1.el7", "kernel-3.10.0-1160.83.1.el7", 1),
         (
-            "4.6~pre16262021g84ef6bd9-3.fc35",
-            "4.6~pre16262021g84ef6bd9-3.fc35",
+            "kernel-core-0:4.6~pre16262021g84ef6bd9-3.fc35",
+            "kernel-core-0:4.6~pre16262021g84ef6bd9-3.fc35",
             0,
         ),
-        ("2:8.2.3568-1.fc35", "2:8.2.3568-1.fc35", 0),
+        ("kernel-core-2:8.2.3568-1.fc35", "kernel-core-2:8.2.3568-1.fc35", 0),
     ),
 )
 def test_compare_package_versions(version1, version2, expected):
     assert pkghandler.compare_package_versions(version1, version2) == expected
 
 
+@pytest.mark.parametrize(
+    ("version1", "version2", "exception_message"),
+    (
+        ("kernel-core-0:390-287.fc36", "kernel-0:390-287.fc36", "kernel-core does not match kernel"),
+        ("kernel-core-0:390-287.fc36.aarch64", "kernel-core-0:391-287.fc36.i86", "aarch64 does not match i86"),
+    ),
+)
+def test_compare_package_versions_warnings(version1, version2, exception_message):
+
+    with pytest.raises(ValueError, match=exception_message):
+        pkghandler.compare_package_versions(version1, version2)
+
+
+def test_parse_pkg_string_dnf(monkeypatch):
+
+    pkg = "kernel-core-0:390-287.fc36.aarch64"
+    parse_pkg_with_dnf_mock = mock.Mock()
+    monkeypatch.setattr(pkghandler, "_parse_pkg_with_dnf", value=parse_pkg_with_dnf_mock)
+    monkeypatch.setattr(pkgmanager, "TYPE", "dnf")
+    pkghandler.parse_pkg_string(pkg)
+    parse_pkg_with_dnf_mock.assert_called_once()
+
+
+def test_parse_pkg_string_yum(monkeypatch):
+
+    pkg = "kernel-core-0:390-287.fc36.aarch64"
```

Comment:
I feel that this could be converted to a parametrization, and you could include all possible formats too.